operators/src/processing/meteosat/radiance.rs

```diff
@@ -0,0 +1,502 @@
+        Self {
+            source,
+            no_data_value,
+            offset_key: RasterPropertiesKey {
+                domain: Some("msg".into()),
+                key: "CalibrationOffset".into(),
+            },
+            slope_key: RasterPropertiesKey {
+                domain: Some("msg".into()),
+                key: "CalibrationSlope".into(),
+            },
+        }
+    }
+}
+
+#[async_trait]
+impl<Q, P> QueryProcessor for RadianceProcessor<Q, P>
+where
+    Q: QueryProcessor<Output = RasterTile2D<P>, SpatialBounds = SpatialPartition2D>,
+    P: Pixel,
+{
+    type Output = RasterTile2D<PixelOut>;
+    type SpatialBounds = SpatialPartition2D;
+
+    async fn query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<BoxStream<'a, Result<Self::Output>>> {
+        let src = self.source.query(query, ctx).await?;
+        Ok(src
+            .map(move |tile| match tile {
+                Ok(tile) if tile.grid_array.is_empty() => Ok(RasterTile2D::new_with_properties(
+                    tile.time,
+                    tile.tile_position,
+                    tile.global_geo_transform,
+                    EmptyGrid::new(tile.grid_array.grid_shape(), self.no_data_value).into(),
+                    tile.properties,
+                )),
+                Ok(tile) => {
+                    let mg = tile.grid_array.into_materialized_grid();
+
+                    let mut out = Grid2D::new(
+                        mg.grid_shape(),
+                        vec![self.no_data_value; mg.data.len()],
+                        Some(self.no_data_value),
+                    )
+                    .expect("raster creation must succeed");
+
+                    let offset = f64::try_from(
+                        tile.properties
+                            .properties_map
+                            .get(&self.offset_key)
+                            .ok_or(crate::error::Error::MissingRasterProperty {
+                                property: "msg.CalibrationOffset".into(),
+                            })?
+                            .clone(),
+                    )? as PixelOut;
+                    let slope = f64::try_from(
+                        tile.properties
+                            .properties_map
+                            .get(&self.slope_key)
+                            .ok_or(crate::error::Error::MissingRasterProperty {
+                                property: "msg.CalibrationSlope".into(),
+                            })?
+                            .clone(),
+                    )? as PixelOut;
+
+                    let tgt = &mut out.data;
+
+                    for (idx, v) in mg.data.iter().enumerate() {
+                        if mg.no_data_value.map_or(true, |ndv| ndv != *v) {
+                            let val: PixelOut = (*v).as_();
+                            tgt[idx] = offset + val * slope;
+                        }
```

Comment:
I recall that we have an `is_no_data` method on `Pixel`

Reply:
Fixed in: 3be3fd383500c6069bb49ffde1523b83a9c4b184